test/test.ts

```diff
@@ -1,35 +1,178 @@
 import chai from "chai";
 import chaiAsPromised from "chai-as-promised";
-import { getParsedData } from "backend-class";
-import fsCases from "./cases-fs";
-import axiosCases from "./cases-axios";
+import {
+  DataServer,
+  getBodyElement,
+  getParsedHref,
+  getParsedURL,
+  parseTree
+} from "backend-class";
+import casesDataParserURL from "./cases-data-parser/cases-url";
+import casesDataParserHREF from "./cases-data-parser/cases-href";
+import casesDataParserHTTP from "./cases-data-parser/cases-axios";
+import casesDataParserFS from "./cases-data-parser/cases-fs";
+import casesDataServerHTTP from "./cases-data-server/cases-axios";
+import casesDataServerFS from "./cases-data-server/cases-fs";
 // Note: baseUrl is set to src/ https://stackoverflow.com/a/43721170
 
 // Note: test async with chai-as-promised https://stackoverflow.com/a/45496509
 chai.use(chaiAsPromised);
 
 describe("Data parser test", () => {
+  // Test URL parsers
+  describe("Test URL parser", () => {
+    // Positive cases
+    casesDataParserURL.forEach(([desc, url, href, file]) => {
+      it(desc!, () => {
+        const parsedURL = getParsedURL(url!);
+        chai.assert.equal(parsedURL.url, href);
+        chai.assert.equal(parsedURL.filepath, file);
+      });
+    });
+
+    // Negative cases
+    it("Empty URL", () => chai.expect(() => getParsedURL("")).to.throw());
+    it("Starts with slash character in URL", () =>
+      chai.expect(() => getParsedURL("/")).to.throw());
+    it("Starts with colon character in URL", () =>
+      chai.expect(() => getParsedURL(":web server")).to.throw());
+  });
+
+  // Test anchor href URL parser
+  describe("Test href URL parser", () => {
+    // Positive cases
+    casesDataParserHREF.forEach(([desc, url, href, result]) => {
+      it(desc, () => {
+        const parsedURL = getParsedURL(url);
+        chai.assert.equal(getParsedHref(parsedURL, href), result);
+      });
+    });
+
+    // Negative cases
+    // it("Empty URL", () => {
+    //   const parsedURL = getParsedURL("");
+    //   chai.assert.equal(getParsedHref(parsedURL, "anything"), "");
+    // });
+  });
```

Comment:
Does this test not work currently? Not sure if we should be committing commented code

Reply:
o oops i forgor 💀